x-pack/plugin/core/src/main/resources/slm-history.json

```diff
@@ -0,0 +1,61 @@
+{
+  "index_patterns": [
+    ".slm-history-${xpack.slm.template.version}*"
+  ],
+  "order": 2147483647,
+  "settings": {
+    "index.number_of_shards": 1,
+    "index.number_of_replicas": 0,
+    "index.auto_expand_replicas": "0-1",
+    "index.lifecycle.name": "slm-history-ilm-policy",
+    "index.format": 1
+  },
+  "mappings": {
+    "_doc": {
+      "_meta": {
+        "slm-history-version": "${xpack.slm.template.version}"
+      },
+      "dynamic": false,
```

Comment:
I think setting this to `strict` would make it easier to catch bugs in our tests

Reply:
It's true that it would make it easier to catch bugs in our tests, but the reason I don't want to set `strict` in this mapping is that:
1) Looking at `CreateSnapshotRequest`, there are some additional options that don't appear to be commonly used and I don't totally understand. For example, the request object [parses out a `settings` field](https://github.com/elastic/elasticsearch/blob/5b97fffb016f9cd5ec88c75711c2f2ad34e6a773/server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotRequest.java#L400-L404) which doesn't appear to be currently used, and [`ignore_unavailable` is actually part of an `IndicesOptions` class that has a number of options](https://github.com/elastic/elasticsearch/blob/2d6c46255cd3f642ddbf18fa55e2fad194fc01b6/server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java#L324-L331) available, which may or may not be in use for snapshots. I don't want to run into issues where these change, or an option gets added, and recording history fails just because we didn't add the field to the mapping (especially since those fields will probably never get queried and so don't need to be indexed, but may be useful to store in the _source).
2) Once `_meta` for snapshots is merged, users will be able to add that to the `config` section of their policy, which can contain arbitrary keys. We might be able to filter it out for this history record, but the simplicity of just storing the entire configuration is nice.